### ISSUE TITLE
Font issue with Arch based Distro

### ISSUE THREAD

**Issue by RoelDijkstra** [bytepixie/snippetpixie#88]
**Describe the bug**
Font Issue, please refer to the screenshot sent, removing fontcache directory did not solve the issue.

**To Reproduce**
Steps to reproduce the behaviour:

Install Snippet Pixie via Snap Store using Manjaro KDE distro

**Expected behaviour**

Normal text

**Screenshots**

https://i.imgur.com/n9SSSpc.png

**Desktop (please complete the following information):**
 - OS: Manjaro Linux
 -  Snippet Pixie Version [e.g. 1.0] = newest snap install

**Comment by ianmjones** [comment]
@RoelDijkstra Do you have the same problem with any other snaps?

I see there's been general problems with Manjaro/Arch and fonts with snaps in the past, e.g:

https://forum.snapcraft.io/t/snap-store-fonts-on-arch-linux-are-merely-empty-rectangles/15373/8
https://forum.snapcraft.io/t/snapped-app-not-loading-fonts-on-fedora-and-arch/12484/96
https://forum.snapcraft.io/t/snap-store-dont-display-text/12219/18

Were the above forum posts where you got the idea to do something like the following?

```
sudo rm /var/cache/fontconfig/*
rm ~/.cache/fontconfig/*
fc-cache -r
```

**Comment by RoelDijkstra** [comment]
@ianmjones Hey Ian, thanks for the quick response. I only have this issue for the snippetpixie snap, I indeed googled the problem and have tried to remove the caching multiple times. It doesn't seem to resolve the issue. Could you tell me what font the app is using? I will try to install it manually. It shouldn't been necessary because multilib is enabled but it's atleast worth the try.

**Comment by ianmjones** [comment]
Thanks for the extra info @RoelDijkstra.

> Could you tell me what font the app is using?

Hmm, I don't think Snippet Pixie requires any particular font, or at least it shouldn't, the default system font is used as provided by the active snap theme.

Let's see if there's any debug messages that can help us work out what's missing. Please can you run through the following experiment?

It's easiest if you have 2 terminal windows open....

1. In your 1st terminal, run `snippetpixie --stop`
1. Run `export G_MESSAGES_DEBUG=all`
1. Run `snippetpixie`
1. In the 2nd terminal window, fully close Snippet Pixie after it displays by running `snippetpixie --stop`.

Please then copy and paste the output from the 1st terminal window here, hopefully there will be complaints about missing fonts etc.

**Comment by RoelDijkstra** [comment]
Alright, here we go:

`snippetpixie
/usr/lib/gio/modules/libgiolibproxy.so: undefined symbol: g_task_set_name
Failed to load module: /usr/lib/gio/modules/libgiolibproxy.so
/usr/lib/gvfs/libgvfscommon.so: undefined symbol: g_task_set_name
Failed to load module: /usr/lib/gio/modules/libgvfsdbus.so
/var/lib/snapd/snap/snippetpixie/98/usr/bin/../lib/x86_64-linux-gnu/libgnutls.so.30: version `GNUTLS_3_6_3' not found (required by /usr/lib/gio/modules/libgiognutls.so)
Failed to load module: /usr/lib/gio/modules/libgiognutls.so
/usr/lib/gio/modules/libgiognomeproxy.so: undefined symbol: g_task_set_name
Failed to load module: /usr/lib/gio/modules/libgiognomeproxy.so
/usr/lib/gvfs/libgvfscommon.so: undefined symbol: g_task_set_name
Failed to load module: /usr/lib/gio/modules/libgioremote-volume-monitor.so

(process:2121630): GLib-GObject-WARNING **: 17:09:13.442: cannot register existing type 'DConfSettingsBackend'

(process:2121630): GLib-CRITICAL **: 17:09:13.442: g_once_init_leave: assertion 'result != 0' failed

(process:2121630): GLib-GIO-WARNING **: 17:09:13.442: Tried to register an extension of the type (null) to extension point gsettings-backend. Expected type is GObject.
Gtk-Message: 17:09:13.443: Failed to load module "canberra-gtk-module"
Gtk-Message: 17:09:13.443: Failed to load module "xapp-gtk3-module"
Gtk-Message: 17:09:13.484: Failed to load module "colorreload-gtk-module"
Gtk-Message: 17:09:13.484: Failed to load module "window-decorations-gtk-module"

** (com.github.bytepixie.snippetpixie:2121630): WARNING **: 17:09:13.485: CustomShortcutSettings.vala:31: Schema "org.gnome.settings-daemon.plugins.media-keys" is not installed on your system.

** (com.github.bytepixie.snippetpixie:2121630): CRITICAL **: 17:09:13.485: custom_shortcut_settings_list_custom_shortcuts: assertion 'available' failed

** (com.github.bytepixie.snippetpixie:2121630): CRITICAL **: 17:09:13.485: custom_shortcut_settings_create_shortcut: assertion 'available' failed

** (com.github.bytepixie.snippetpixie:2121630): WARNING **: 17:09:13.513: CustomShortcutSettings.vala:31: Schema "org.gnome.settings-daemon.plugins.media-keys" is not installed on your system.

** (com.github.bytepixie.snippetpixie:2121630): CRITICAL **: 17:09:13.513: custom_shortcut_settings_list_custom_shortcuts: assertion 'available' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.171: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.171: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.171: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.171: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.171: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.171: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.171: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.172: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.172: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.172: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.172: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.172: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

(com.github.bytepixie.snippetpixie:2121630): GLib-GObject-CRITICAL **: 17:09:30.173: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
`

**Comment by ianmjones** [comment]
@RoelDijkstra Whoa, ok, did not expect that kind of output!

As mentioned in https://forum.snapcraft.io/t/snapped-app-not-loading-fonts-on-fedora-and-arch/12484/ I've just rebuilt and released the snippetpixie snap to be sure it is using the GNOME font stuff. Hopefully that'll fix things for you.

To ensure your version of snippetpixie is properly refreshed and connected to the GNOME snap, please run `snippetpixie --stop`, then `sudo snap remove snippetpixie`, then `sudo snap install snippetpixie --classic`.

If that does not help with the font issue when you then just run Snippet Pixie normally, please send me the output from `snap list`.

**Comment by RoelDijkstra** [comment]
Indeed! I started using your application on Ubuntu and it worked perfectly when installed using the snap store. This should be some Arch based problem, probably missing some dependency or package. I use Pixie for client work all the time, productivity booster :)  Here is my snap list:

snap list
Name                Version                     Rev    Tracking       Publisher     Notes
code                ea3859d4                    52     latest/stable  vscode✓       classic
core                16-2.48.2                   10583  latest/stable  canonical✓    core
core18              20201210                    1944   latest/stable  canonical✓    base
core20              20201210                    904    latest/stable  canonical✓    base
drawio              14.1.8                      99     latest/stable  jgraph✓       -
glow                1.2.1                       32     latest/stable  tbmb          -
gnome-3-28-1804     3.28.0-19-g98f9e67.98f9e67  145    latest/stable  canonical✓    -
gtk-common-themes   0.1-50-gf7627e4             1514   latest/stable  canonical✓    -
heroku              v7.47.6                     4017   latest/stable  heroku✓       classic
insomnia            2020.5.2                    118    latest/stable  gschier1990   -
mattermost-desktop  4.6.1                       295    latest/stable  snapcrafters  -
nmap                7.91                        2061   latest/stable  maxiberta     -
phpstorm            2020.3.1                    197    latest/stable  jetbrains✓    classic
postman             7.36.1                      130    latest/stable  postman-inc✓  -
slack               4.12.0                      34     latest/stable  slack✓        classic
snapd               2.48.2                      10707  latest/stable  canonical✓    snapd
snippetpixie        1.5.0                       101    latest/stable  bytepixie     classic
termius-app         7.3.2                       61     latest/stable  termius✓      -
tusk                0.23.0                      29     latest/stable  klaussinani   -

If you need something else, let me know!

Regards,
Roel

**Comment by ianmjones** [comment]
@RoelDijkstra thanks for the list of installed snaps, versions look as expected.

Please can you tell me which version of Manjaro you're using (which version was originally installed), and which desktop environment or window manager you're currently using? I'll look into spinning up a clean install with the same versions and try to reproduce the problem.

I take it you've updated your packages and rebooted a few times recently and checked Snippet Pixie after that?

**Comment by RoelDijkstra** [comment]
@ianmjones I've checked this out and my Manjaro version is 20.2.1
, I am currently using KDE Plasma Manjaro, so no Window Managers like i3, bspwm. I am using a KWin script named Krohnkite to manage my windows. I did indeed reboot and updated my packages frequently.

**Comment by RoelDijkstra** [comment]
@ianmjones Any Updates?

**Comment by ianmjones** [comment]
@RoelDijkstra Sorry, life got in the way, thanks for the ping.

Just reproduced it with a clean install of Manjaro KDE, but I haven't got a clue why it's not working as all the fonts etc seem to be in place and system/user/snap fontconfig caches are re-created after purging. :shrug: 

I'll keep looking as time permits.

**Comment by ianmjones** [comment]
I'm at a loss, only Manjaro seems to have this issue.

So I've asked in the Manjaro forum for some advice on this issue, with luck someone can help solve this mystery. :crossed_fingers: 

https://forum.manjaro.org/t/advice-for-a-developer-on-how-to-fix-font-issues-with-snap/54299

**Comment by ianmjones** [comment]
@RoelDijkstra I think I've fixed compatibility with Manjaro! :tada: 

![snippetpixie-running-on-manjaro-kde-20_2_1](https://user-images.githubusercontent.com/4710/109396932-f53e7d80-792b-11eb-9f94-3b73d0146f6d.png)

If you still have Snippet Pixie installed, please run `sudo snap refresh snippetpixie --candidate` to get it.

If you have uninstalled Snippet Pixie, please run `sudo snap install snippetpixie --classic --candidate` to get it.

If using KDE the built in shortcut setting seems to not save, so please create a global custom shortcut to get the search and paste window by assigning it the command `snippetpixie --search-and-paste`.

Please let me know how you get on.

**Comment by RoelDijkstra** [comment]
Sick, it is indeed working as expected and I was just about to say something about the shortcut. 
Thanks for spending this time on fixing the issue for me, I really appreciate that!
This will boost my work productivity for sure :)

**Comment by ianmjones** [comment]
Fixed in 1.5.1, released 4 months ago! :facepalm:
